terraform/modules/gro_ingestion_service/lambdas/src/functions/splitXml.function.ts

```diff
@@ -0,0 +1,35 @@
+import { Handler, S3Event } from "aws-lambda"
+import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3"
+import { XMLParser } from "fast-xml-parser"
+
+const client = new S3Client({})
+const parser = new XMLParser()
+
+const getGroFile = async (event: S3Event) => {
+
+    const getGroCommand = new GetObjectCommand({
+        Bucket: event.Records[0].s3.bucket.name,
+        Key: event.Records[0].s3.object.key
+    })
+    const groFileResponse = await client.send(getGroCommand)
+
+    return groFileResponse.Body?.transformToString()
+}
+
+export const handler: Handler = async (event: S3Event) => {
+    const groXml = await getGroFile(event)
+
+    if (!groXml) {
+        const logParams = {
+            fileKey: event.Records[0].s3.object.key,
+            error: `File with key ${event.Records[0].s3.object.key} not found`,
+        }
+        console.error("Failed to insert records into DynamoDB", logParams)
+        return {
+            statusCode: 404,
+        }
+    }
+
+    const groJson = parser.parse(groXml)
+    return { deathRegistrations: groJson.deathRegistrationGroup }
```

Comment:
Not sure if we now want to add a status code here?

Reply:
This is all happening inside lambdas from S3 to DynamoDB, I don't think we need a status code